docs/examples/tutorial_signal_processing.py

```diff
@@ -0,0 +1,286 @@
+import requests
+
+import pynapple as nap
+
+# %%
+# ***
+# Downloading the data
+# ------------------
+# First things first: Let's download and extract the data
+path = "data/A2929-200711"
+extract_to = "data"
+if extract_to not in os.listdir("."):
+    os.mkdir(extract_to)
+if path not in os.listdir("."):
+    # Download the file
+    response = requests.get(
+        "https://www.dropbox.com/s/su4oaje57g3kit9/A2929-200711.zip?dl=1"
+    )
+    zip_path = os.path.join(extract_to, "/downloaded_file.zip")
+    # Write the zip file to disk
+    with open(zip_path, "wb") as f:
+        f.write(response.content)
+    # Unzip the file
+    with ZipFile(zip_path, "r") as zip_ref:
+        zip_ref.extractall(extract_to)
+
+
+# %%
+# ***
+# Parsing the data
+# ------------------
+# Now that we have the data, we must append the 2kHz LFP recording to the .nwb file
+eeg_path = "data/A2929-200711/A2929-200711.dat"
+frequency = 20000  # Hz
+n_channels = 16
+f = open(eeg_path, "rb")
+startoffile = f.seek(0, 0)
+endoffile = f.seek(0, 2)
+f.close()
+bytes_size = 2
+n_samples = int((endoffile - startoffile) / n_channels / bytes_size)
+duration = n_samples / frequency
+interval = 1 / frequency
+fp = np.memmap(eeg_path, np.int16, "r", shape=(n_samples, n_channels))
+timestep = np.arange(0, n_samples) / frequency
+eeg = nap.TsdFrame(t=timestep, d=fp)
+nap.append_NWB_LFP("data/A2929-200711/", eeg)
+
+
+# %%
+# Let's save the RoiResponseSeries as a variable called 'transients' and print it
+FS = 1250
+# data = nap.load_file("data/A2929-200711/pynapplenwb/A2929-200711.nwb")
+data = nap.load_file("data/stable.nwb")
+print(data["ElectricalSeries"])
+
+# %%
+# ***
+# Selecting slices
+# -----------------------------------
+# Let's consider a two 1-second slices of data, one from the sleep epoch and one from wake
+NES = nap.TsdFrame(
+    t=data["ElectricalSeries"].index.values,
+    d=data["ElectricalSeries"].values,
+    time_support=data["ElectricalSeries"].time_support,
+)
+wake_minute = NES.value_from(NES, nap.IntervalSet(900, 960))
+sleep_minute = NES.value_from(NES, nap.IntervalSet(0, 60))
+
+# %%
+# ***
+# Plotting the LFP activity of one slices
+# -----------------------------------
+# Let's plot
+fig, ax = plt.subplots(2)
+for channel in range(sleep_minute.shape[1]):
+    ax[0].plot(
+        sleep_minute.index.values,
+        sleep_minute[:, channel],
+        alpha=0.5,
+        label="Sleep Data",
+    )
+ax[0].set_title("Sleep ephys")
+for channel in range(wake_minute.shape[1]):
+    ax[1].plot(
+        wake_minute.index.values, wake_minute[:, channel], alpha=0.5, label="Wake Data"
+    )
+ax[1].set_title("Wake ephys")
+plt.show()
+
+
+# %%
+# There is much shared information between channels, and wake and sleep don't seem visibly different.
+# Let's take the Fourier transforms of one channel for both and see if differences are present
+channel = 1
+fig, ax = plt.subplots(1)
+fft_sig, fft_freqs = nap.compute_spectrum(sleep_minute[:, channel], fs=int(FS))
```

Comment:
This should be a pandas data frame so you can only return a single object. If you pass a TsdFrame, you can get a data frame where every column is one channel and the index are the `fft_freqs`.